### DISCUSSION TITLE
Business Central integration

### DISCUSSION THREAD

**Discussion by Lrss** [saleor/saleor#10718] [Ideas]
I have a customer that uses Dynamics 365 Business Central, and I would like to create an integration for Saleor that can synchronize  stock, orders, and customers. I have a few questions:
- Are there any efforts already that I should be aware of, or should I just begin creating?
- Are there any pointers, designs, or examples that I seek inspiration from?
- Are there an official chat or should I communication via Github discussions/issues/pr?

**Comment by JannikZed** [comment]
hey @Lrss - from our experience, it is very hard to deeply integrate different ecommerce systems with each other. Saleor has everything you need in-place (either live, using webhooks or async using the GraphQL API). But the hard work is trying to bring the different data types together. The more entities you want to sync, the harder it gets. -- do you want to sync stocks, discounts, updates on existing orders, payments etc etc .. you name it :D 

We could assist you in that, depending on the business case behind it. We started to create a universal, ecommerce data hub that already integrates fully with saleor and other ecom systems that we are planning to publish as SaaS. If you are interested in that, just drop me a line somewhere. 

As far as I know, the saleor team tries to focus on the main services, leaving every integration to 3rd party developers using the great app framework, that you can find here: https://docs.saleor.io/docs/3.x/developer/extending/apps/key-concepts

**Reply by Lrss** [reply]
Thanks for the reply. For clarification; Business Central is an ERP system not an E-commerce systems.

The minimal viable product would be to create a Saleor App that could pull inventory quantity from business central in addition to price and discount from a specified price list. Then the seller would have to manually invoice the orders in business central when the receive an order email from Saleor.

Planned features that would eventually be developed are:
- Create customers in Business Central when users registers on Saleor 
- Create orders in Business Central when they are placed in Saleor
- Synchronize shipping status and order status when handled in Business Central
- Create products and pull specifications from Business Central
  - Images
  - Description texts
  - Categories
  - Tags

I have experience pulling all this information from business central in other applications I maintain, and I'm excited to implement it for an E-commerce platform, and Saleor seems like the most open modern candidate.

If I have further questions during development should I just write in this discussion, or are there other preferred channels?

**Reply by Peyo-Supp** [reply]
Hey @Lrss what road did you end taking ? I have a client with very similar requirements

**Reply by getrav** [reply]
Bump. I have the same requirement. I will start building a connector next week. Stay tuned...
